### DISCUSSION TITLE
how to use python sdk WorkflowServiceApi.watch_events api?

### DISCUSSION THREAD

**Discussion by XinSean** [argoproj/argo-workflows#8318] [Q&A]
Hi. i want to watch workflow events, for example, `WorkflowRunning` and `WorkflowSucceeded` events. i use argo workflow python sdk, but can't watch any event, my code is

> from pprint import pprint
> 
> import argo_workflows
> from argo_workflows.api import workflow_service_api
> from argo_workflows.model.container import Container
> from argo_workflows.model.io_argoproj_workflow_v1alpha1_template import IoArgoprojWorkflowV1alpha1Template
> from argo_workflows.model.io_argoproj_workflow_v1alpha1_workflow import IoArgoprojWorkflowV1alpha1Workflow
> from argo_workflows.model.io_argoproj_workflow_v1alpha1_workflow_spec import \
>     IoArgoprojWorkflowV1alpha1WorkflowSpec
> from argo_workflows.model.object_meta import ObjectMeta
> 
> configuration = argo_workflows.Configuration(host="http://192.168.10.100:2746")
> configuration.verify_ssl = False
> 
> manifest = IoArgoprojWorkflowV1alpha1Workflow(
>     metadata=ObjectMeta(generate_name='hello-world-'),
>     spec=IoArgoprojWorkflowV1alpha1WorkflowSpec(
>         entrypoint='whalesay',
>         templates=[
>             IoArgoprojWorkflowV1alpha1Template(
>                 name='whalesay',
>                 container=Container(
>                     image='docker/whalesay:latest', command=['cowsay'], args=['hello world']))]))
> api_client = argo_workflows.ApiClient(configuration)
> api_instance = workflow_service_api.WorkflowServiceApi(api_client=api_client)
> 
> namespace = "argo-workflow"
> api_response = api_instance.watch_events(namespace, async_req=True)
> pprint(api_response.get())
> 

Thank you very much for any help

**Comment by terrytangyuan** [comment]
Did you see any error messages?

**Comment by alexec** [comment]
This API does not work because OpenAPI does not support Server Sent Events
